lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -681,9 +684,37 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
 
         alert_rules_as_dict = alert_rules.as_dict()
 
+        alert_rules_as_dict = self._parse_and_add_extra_alerts(alert_rules_as_dict)
+
         if alert_rules_as_dict:
             relation.data[self._charm.app]["alert_rules"] = json.dumps(alert_rules_as_dict)
 
+    def _parse_and_add_extra_alerts(self, alert_rules_as_dict: dict) -> dict:
```

Comment:
- Should we refactor this into a classmethod?
- Tests would be nice.

Reply:
Why would it be a classmethod? It's not a factory returning an instance of the class. It could be static, but so could a lot of things in the codebase.

Reply:
yes, this could be static. However, what is the scenario where you will use it outside of instance?

Reply:
It's not a big deal. As Ryan says we have a lot of this in the codebase.
I meant to better convey intent: is this better as a free function or static?

(In the past we had a debate about staticmethod vs classmethod and after being provided with a [reference](https://mail.python.org/pipermail/python-ideas/2016-July/041189.html) for Guido saying staticmethod was a mistake we said internally we'd always do classmethod.)

Reply:
how I understand static method is that `it is a function that can work without a class instance and attached to a class just to show where it logically belongs to`

in our case, yes, we can rework it to static method or even to a separate function. But what the use case will be?

Reply:
Ideally I'd like to have completely separate code for manipulating those data structures.
Augmenting a `groups` yaml is outside the responsibility of the charm class, imo.
It would then also be easier to test.
But this would probably be a big refactoring and outside the scope of this PR.
Resolving.

Reply:
Even if Guido thinks it was a mistake, plenty of others have found good uses for it with `@staticmethod` members on classes. It really doesn't make sense on a private method though.

`@classmethod` is really useful for factories, or accessing class attributes from subclasses (or other). This code isn't returning an instance of the class, and it isn't accessing class-level variables (which is only `on` anyway, and actually requires a "real" instance due to the way OF dispatches things to object handles), so it doesn't have a good reason to be a classmethod regardless.